### PR TITLE
Arranging BMO & Ironic deployments with separate folders 

### PR DESCRIPTION
**Pull request by furkatgofurov7**
**What this PR does / why we need it:**
This PR breaks [metal3-io/baremetal-operator#481](https://github.com/metal3-io/baremetal-operator/pull/481) into two parts and this is a **first** part which introduces what is needed to be added in that PR. It structures `deploy` folder as simple as possible (only for BMO) and fits for operator-sdk while all ironic part is moved to its own folder.

Related to #540 
Follows the proposal in [metal3-docs/pull/91](https://github.com/metal3-io/metal3-docs/pull/91)
Documented in #541

### CODE REVIEW

**Comment by dhellmann** [comment]
Ironic and the operator used to be in a single deployment so that the operator could use localhost to talk to ironic without any authentication. Why do we want them to be in separate deployments now?

**Comment by furkatgofurov7** [comment]
The reason is that we want to deploy BMO on its own in CAPM3 and Ironic independently. The use case explaining what we are trying to achieve is a pivoting work and these changes will make it possible to make BMO as part of metal3 in clusterctl.

**Comment by furkatgofurov7** [comment]
> this looks good. We should probably however, add a lot of documentation, and either add a top-level kustomization that will deploy BMO + ironic or add instructions with command examples on how to achieve this.

Yes sure, I will add instructions with commands in `docs` folder, if it is right place to do so?

**Comment by maelk** [comment]
you should probably still put something in the README, to refer to the docs, or a simpler version of the docs, like the tldr of the docs

**Comment by furkatgofurov7** [comment]
> this looks good. We should probably however, add a lot of documentation, and either add a top-level kustomization that will deploy BMO + ironic or add instructions with command examples on how to achieve this.

I have added instructions with command examples in this [PR#541](https://github.com/metal3-io/baremetal-operator/pull/541)

**Comment by furkatgofurov7** [comment]
> you should probably still put something in the README, to refer to the docs, or a simpler version of the docs, like the tldr of the docs

I have modified the previous configuration file and referral link to configuration document inside it

**Comment by furkatgofurov7** [comment]
@dhellmann if it looks good to you as well, it needs your approval to be merged.

File: deploy/default/kustomization.yaml [hunk 1]
```diff
@@ -5,13 +5,8 @@ resources:
 - ../namespace
 - ../rbac
 - ../crds
-- ../operator/ironic
+- ../operator
```
Comment (maelk) [inline deploy/default/kustomization.yaml:8]: we should add ironic here to not modify the current behaviour

Comment (furkatgofurov7) [reply deploy/default/kustomization.yaml:8]: Ah true, added. Thanks!

**Pull request merged**